ocs_sample_library_preview/SDS/SdsType.py

```diff
@@ -1,193 +1,353 @@
+from __future__ import annotations  # To type hint the enclosing class
 import json
-from .SdsTypeCode import SdsTypeCode
+
+from .SdsExtrapolationMode import SdsExtrapolationMode
+from .SdsInterpolationMode import SdsInterpolationMode
+from .SdsTypeCode import SdsTypeCode as SdsTypeCodeType
 from .SdsTypeProperty import SdsTypeProperty
 
 
 class SdsType(object):
     """Sds type definitions"""
 
-    def __init__(self, id=None, name=None, description=None, baseType=None,
-                 sdsTypeCode=SdsTypeCode.Empty, properties=[]):
```

Comment:
keep the default?  since it is mandatory this in most cases

Reply:
It is indeed a required parameter, but I felt like specifying a default effectively made it not required and might result in unexpected behavior if someone simply forgot to specify it. Personally I would prefer to get an error if I forgot a required parameter, rather than default to 'Empty', then I have to delete it to try again. 
Open to discussion if you disagree, but this was a conscious decision.

Reply:
It is indeed a required parameter, but I felt like specifying a default effectively made it not required and might result in unexpected behavior if someone simply forgot to specify it. Personally I would prefer to get an error if I forgot a required parameter, rather than default to 'Empty', then I have to delete it to try again. 
Open to discussion if you disagree, but this was a conscious decision.